workers/unity/Assets/Fps/Scripts/GameLogic/VisibilityAndCollision.cs

```diff
@@ -22,11 +22,23 @@ private void Awake()
         private void OnEnable()
         {
             health.HealthUpdated += HealthUpdated;
-            SetIsVisible(health.Data.Health > 0);
+            UpdateVisibility();
         }
 
-        private void SetIsVisible(bool visible)
+        private void HealthUpdated(float newHealth)
+        {
+            UpdateVisibility();
+        }
+
+        private void UpdateVisibility()
         {
+            if (health == null)
+            {
+                return;
+            }
```

Comment:
This null check shouldn't ever be true given where this is called - but better safe than sorry I guess 😛